ui/components/src/StatusHistoryChart/StatusHistoryChart.tsx

```diff
@@ -0,0 +1,169 @@
+// Copyright 2024 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Box } from '@mui/material';
+import { HeatmapChart as EChartsHeatmapChart } from 'echarts/charts';
+import {
+  GridComponent,
+  DatasetComponent,
+  TitleComponent,
+  TooltipComponent,
+  VisualMapComponent,
+  LegendComponentOption,
+} from 'echarts/components';
+import { EChartsCoreOption, use } from 'echarts/core';
+import { CanvasRenderer } from 'echarts/renderers';
+import { TimeScale } from '@perses-dev/core';
+import { useChartsTheme } from '../context/ChartsProvider';
+import { useTimeZone } from '../context/TimeZoneProvider';
+import { formatWithTimeZone } from '../utils';
+import { EChart } from '../EChart';
+import { getColorsForValues } from './utils/get-color';
+import { getFormattedStatusHistoryAxisLabel } from './get-formatted-axis-label';
+import { getTooltipPosition } from './utils/get-tooltip-position';
+
+use([
+  EChartsHeatmapChart,
+  VisualMapComponent,
+  GridComponent,
+  DatasetComponent,
+  TitleComponent,
+  TooltipComponent,
+  CanvasRenderer,
+]);
+
+export type StatusHistoryData = [number, number, number | undefined];
+
+export interface StatusHistoryChartProps {
+  height: number;
+  data: StatusHistoryData[];
+  xAxisCategories: number[];
+  yAxisCategories: string[];
+  legend?: LegendComponentOption;
+  timeScale?: TimeScale;
+}
+
+export function StatusHistoryChart(props: StatusHistoryChartProps) {
+  const { height, data, xAxisCategories, yAxisCategories, timeScale } = props;
+  const { timeZone } = useTimeZone();
+  const chartsTheme = useChartsTheme();
+
+  const uniqueValues = [...new Set(data.map((item) => item[2]))].filter(
+    (value): value is number => value !== undefined
+  );
+
+  // get colors from theme and generate colors if not provided
+  const colors = uniqueValues.map((value, index) => ({
+    value,
+    color: getColorsForValues(
+      uniqueValues,
+      Array.isArray(chartsTheme.echartsTheme.color)
+        ? chartsTheme.echartsTheme.color.filter((color): color is string => typeof color === 'string')
+        : []
+    )[index],
+  }));
+
+  const option: EChartsCoreOption = {
+    tooltip: {
+      position: getTooltipPosition,
+      formatter: (params: { data: StatusHistoryData }) => {
```

Comment:
Maybe we could reuse what has been done with TimeSeriesChart, in order to have a similar display?

Reply:
StatsChart:
 ![image](https://github.com/user-attachments/assets/f9454a58-4f92-42a3-bb5a-de1d065e5ecf)
TimeSeriesChart: 
![image](https://github.com/user-attachments/assets/283e8a0c-a2b7-4e39-9ef1-b32e55cbf692)

Reply:
<img width="326" alt="image" src="https://github.com/user-attachments/assets/1d010540-8411-40a7-a715-79301f0e56cc">